internal/engine/experiment/tlsmiddlebox/tlsmiddlebox.go

```diff
@@ -0,0 +1,179 @@
+type Measurer struct {
+	config Config
+}
+
+// ExperimentName implements ExperimentMeasurer.ExperimentName.
+func (m *Measurer) ExperimentName() string {
+	return testName
+}
+
+// ExperimentVersion implements ExperimentMeasurer.ExperimentVersion.
+func (m *Measurer) ExperimentVersion() string {
+	return testVersion
+}
+
+var (
+	// errNoInputProvided indicates you didn't provide any input
+	errNoInputProvided = errors.New("no input provided")
+
+	// errInputIsNotAnURL indicates that input is not an URL
+	errInputIsNotAnURL = errors.New("input is not an URL")
+
+	// errInvalidScheme indicates that the scheme is invalid
+	errInvalidScheme = errors.New("scheme must be tlshandshake or https")
+)
+
+// // Run implements ExperimentMeasurer.Run.
+func (m *Measurer) Run(
+	ctx context.Context,
+	sess model.ExperimentSession,
+	measurement *model.Measurement,
+	callbacks model.ExperimentCallbacks,
+) error {
+	if measurement.Input == "" {
+		return errNoInputProvided
+	}
+	parsed, err := url.Parse(string(measurement.Input))
+	if err != nil {
+		return fmt.Errorf("%w: %s", errInputIsNotAnURL, err.Error())
+	}
+	scheme := parsed.Scheme
+	if scheme != "tlshandshake" && scheme != "https" {
+		return errInvalidScheme
+	}
+	tk := new(TestKeys)
+	measurement.TestKeys = tk
+	sni := m.config.sni(parsed.Host)
+	// 1. perform a DNSLookUp
+	outDNS, addrs, err := m.DNSLookup(ctx, parsed.Hostname(), nil)
+	tk.DNSLookUp = outDNS
+	if err != nil {
+		return err
+	}
+	// 2. measure addresses
+	m.MeasureAddrs(ctx, addrs, parsed.Port(), sni, tk)
+	return nil
+}
+
+// MeasureAddrs measures the array of addresses obtained from DNSLookUp
+func (m *Measurer) MeasureAddrs(ctx context.Context, addrs []string, port string,
+	sni string, tk *TestKeys) {
+	tcpEvents := make(chan *model.ArchivalTCPConnectResult, len(addrs))
+	tlsEvents := make(chan *CompleteTrace, 2*len(addrs))
+	wg := new(sync.WaitGroup)
+	addrs = internal.PrepareAddrs(addrs, port)
+	for _, addr := range addrs {
+		wg.Add(1)
+		go m.MeasureSingleAddr(ctx, addr, sni, tcpEvents, tlsEvents, wg)
+	}
+	wg.Wait()
+	tk.TCPConnect = GetTCPEvents(tcpEvents)
+	tk.TLSTrace = GetTLSEvents(tlsEvents)
+}
+
+// MeasureSingleAddr measures a single address
+func (m *Measurer) MeasureSingleAddr(ctx context.Context, addr string,
+	sni string, tcpEvents chan<- *model.ArchivalTCPConnectResult,
+	tlsEvents chan<- *CompleteTrace, wg *sync.WaitGroup) error {
+	defer wg.Done()
+	err := m.MeasureTCP(ctx, addr, tcpEvents)
+	if err != nil {
+		return err
+	}
+	m.MeasureTLS(ctx, addr, sni, tlsEvents)
+	return nil
+}
+
+// NewExperimentMeasurer creates a new ExperimentMeasurer.
+func NewExperimentMeasurer(config Config) *Measurer {
+	return &Measurer{config: config}
+}
+
+type SummaryKeys struct {
+	IsAnomaly bool `json:"-"`
+}
+
+// GetSummaryKeys implements model.ExperimentMeasurer.GetSummaryKeys.
+func (m Measurer) GetSummaryKeys(measurement *model.Measurement) (interface{}, error) {
+	return SummaryKeys{IsAnomaly: false}, nil
```

Comment:
I think you should also start fleshing out the logic which should be used to understand if filtering is happening. This can be very helpful to ensure all relevant test metadata is being collected.